dbt/include/spark/macros/materializations/incremental.sql

```diff
@@ -13,7 +13,7 @@
   {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,  type='table') -%}
 
   {%- set full_refresh = flags.FULL_REFRESH == True and old_relation is not none -%}
-  {%- set type = spark_get_relation_type(this) if old_relation else none -%}
+  {%- set type = old_relation.type if old_relation.type is not none else spark_get_relation_type(this) -%}
```

Comment:
`old_relation.type` should never be `none`, right? I think the failure mode here could be if `old_relation` itself is `none`, in which case we wouldn't want to access it's `type` attribute.

How would you feel about removing this line and changing the line below to:
```
{%- set old_relation_is_view = old_relation is not none and old_relation.type == 'view' -%}
```

This feels a little simpler, plus it will save an introspective query :)

Reply:
Good call, I was hedging against myself